graphql/schemabuilder/input.go

```diff
@@ -133,6 +133,10 @@ func (sb *schemaBuilder) getStructObjectFields(typ reflect.Type) (*graphql.Input
 		}
 		if fieldInfo.OptionalInputField {
 			parser = wrapWithZeroValue(parser)
+			// Make sure the "fieldArgType" we expose in graphQL is a Nullable field.
+			if f, ok := fieldArgTyp.(*graphql.NonNull); ok {
```

Comment:
Other option is to add the fieldArgTyp to the wrapWithZeroValue function as well.